hooks/useDebounce.ts

```diff
@@ -9,12 +9,11 @@ export function useDebounce<T extends (...params: never[]) => void>(milliseconds
     // @see https://github.com/facebook/react/issues/16956#issuecomment-536636418
     fnRef.current = fn;
 
-    const timer = useRef<number>();
+    const timer = useRef<number>(null);
 
-    // eslint-disable-next-line react-hooks/exhaustive-deps
     return useCallback<T>(
         ((...params) => {
-            clearTimeout(timer.current);
+            clearTimeout(timer.current ?? undefined);
```

Comment:
I see we have a library with utility hooks like this: `@react-hookz/web`. Probably we should use `useDebouncedCallback` from there. It handles additional cases like cancelling function execution on unmount.

Reply:
https://github.com/prezly/theme-nextjs-bea/pull/1213/commits/b760c5bd93353dcb5c0300e38ac07d2c9e83ac35